### PR TITLE
Add ipv6 support

### LINKED ISSUE #86: Allow for extra 'listen' directives to enable ipv4/ipv6 simultaneously
My default http and https nginx configs contain:

```
  listen 80;
  listen [::]:80;
```
and 
```
  listen 443 ssl http2;
  listen [::]:443 ssl http2;
```

But the role template currently only allows a single `listen` directive. You can of course sneak a 2nd one in with the `extra_parameters` option, but I assume we're going to be supporting ipv4/ipv6 simultaneously for many years to come

### PR DESCRIPTION
**Pull request by robbyoconnor**
Fixes #86 

Add a listen directive for IPv6

### CODE REVIEW

**Comment by robbyoconnor** [comment]
Fixing broken tests

**Comment by robbyoconnor** [comment]
Stupid missing semi-colons :slightly_smiling_face:

**Comment by robbyoconnor** [comment]
@geerlingguy :point_up:

File: templates/vhost.j2 [hunk 1]
```diff
@@ -2,6 +2,7 @@
 {% if item.server_name_redirect is defined %}
 server {
     listen       {{ item.listen | default('80') }};
+    listen [::]:{{item.listen | default('80') }};
```
Comment (LukeHandle) [inline templates/vhost.j2]: Should be spaced as     `listen       [::]:{{item.listen | default('80') }};` to follow the indent in that block

Comment (robbyoconnor) [reply templates/vhost.j2]: Addressed this.

**Comment by robbyoconnor** [comment]
@geerlingguy bumping this

**Comment by LukeHandle** [comment]
I wonder if there a scenario where these still get reviewed, or if it's a donation with a "I appreciate what you do/have done. If in the future, you get a mo, take a look here please" :male_shrug:

I'm not sure if the stalebot gives a sense of falsehope 😂

**Comment by robbyoconnor** [comment]
I'm fine with things sitting

**Comment by geerlingguy** [comment]
Just checking in on this finally (note that the nginx role is one of the most active, but also one that I think would negatively impact the most users if I blindly accepted any PRs ;-)

It looks good to me, the only question I have is could there be any downside to listening on IPv6 (e.g. if you have it explicitly disabled on your server, will this cause any config errors?).

If I do merge it, I think I'd do it in a new major role version, just because any time I mess with changing defaults with IPv4 and IPv6, inevitably some router, proxy, or other server _somewhere_ starts having some obscure issue. (We've had _how many years_ to make that not a problem??).

**Comment by robbyoconnor** [comment]
Just bumping to make @stale happy.

**Comment by robbyoconnor** [comment]
er reopen this please

**Comment by madonius** [comment]
bumping to keep stalebot happy

**Comment by benibr** [comment]
any chance this getting merged soon? I think in 2021 listening on IPv6 is reasonable. Especially if the change comes with major release. Would really appreciate this feature.

Right now I fumble around it with:

```
    extra_parameters: |
      # https://github.com/geerlingguy/ansible-role-nginx/pull/125 PR since 2017 oO
      listen [::]:443 ssl http2;
```

**Comment by robbyoconnor** [comment]
@benibr could just pull from my github branch: 

```yaml
- name: nginx # remove when geerlingguy/ansible-role-nginx#125 is merged
  src: https://github.com/robbyoconnor/ansible-role-nginx.git
  version: add-ipv6-support
```

File: templates/vhost.j2 [hunk 2]
```diff
@@ -2,6 +2,7 @@
 {% if item.server_name_redirect is defined %}
 server {
     listen       {{ item.listen | default('80') }};
+    listen       [::]:{{item.listen | default('80') }};
```
Comment (geerlingguy) [inline templates/vhost.j2:5]: I think before merging this, I would like this to be behind a flag (and we can default to `true`), like:

```yaml
nginx_listen_ipv6: true
```

And then this line would become:

```
{% if nginx_listen_ipv6 %}
    listen       [::]:{{item.listen | default('80') }};
{% endif %}
```

(And the same for the other occurence.)

Along with this change, the new variable would need to be in defaults and also added in the README.

**Review by geerlingguy — CHANGES_REQUESTED** [review]

File: templates/vhost.j2 (hunk 2 above)
Comment (robbyoconnor) [reply templates/vhost.j2:5]: Will do -- I'll get this done tomorrow

**Comment by robbyoconnor** [comment]
@geerlingguy changes you requested have been made

**Review by geerlingguy — APPROVED** [review]

**Pull request merged**